test/json-schema-serializer-interceptor.spec.ts

```diff
@@ -0,0 +1,322 @@
+import S from 'fluent-schema';
+import * as sinon from 'sinon';
+import * as request from 'supertest';
+import { Controller, Get, Post, INestApplication, Catch } from '@nestjs/common';
+import {
+  ApiBody,
+  ApiQuery,
+  ApiParam,
+  ApiHeader,
+  ApiResponse,
+} from '@nestjs/swagger';
+import { Test } from '@nestjs/testing';
+import { ArgumentsHost, ExceptionFilter } from '@nestjs/common';
+import { VALIDATOR, VALIDATOR_ERROR_HANDLER } from '../lib/constants';
+import {
+  AjvValidator,
+  JsonSchemaSerializerInterceptor,
+  ValidationModule,
+} from '../lib';
+import { expect } from 'chai';
+
+@Catch()
+export class CustomExceptionFilter implements ExceptionFilter {
+  catch(err: Error, host: ArgumentsHost): any {
+    const ctx = host.switchToHttp();
+    const res = ctx.getResponse();
+
+    return res.status(500).json({
+      message: err.message,
+    });
+  }
+}
+
+describe('JsonSchemaSerializerInterceptor', () => {
+  let app: INestApplication;
+
+  async function createApp(controller): Promise<INestApplication> {
+    const moduleRef = await Test.createTestingModule({
+      imports: [ValidationModule.forRoot()],
+      controllers: [controller],
+    }).compile();
+
+    app = moduleRef.createNestApplication();
+    const ajvValidator = app.get(VALIDATOR);
+    const errorHandler = app.get(VALIDATOR_ERROR_HANDLER);
+
+    app.useGlobalInterceptors(
+      new JsonSchemaSerializerInterceptor(ajvValidator, errorHandler),
+    );
+
+    app.useGlobalFilters(new CustomExceptionFilter());
+    await app.init();
+
+    return app;
+  }
```

Comment:
```suggestion
  async function createApp(controller): Promise<INestApplication> {
    const moduleRef = await Test.createTestingModule({
      imports: [ValidationModule.forRoot()],
      controllers: [controller],
    }).compile();

    app = moduleRef.createNestApplication();
    const ajvValidator = app.get(VALIDATOR);
    const errorHandler = app.get(VALIDATOR_ERROR_HANDLER);

    app.useGlobalInterceptors(
      new JsonSchemaSerializerInterceptor(ajvValidator, errorHandler),
    );

    app.useGlobalFilters(new CustomExceptionFilter());
    await app.init();

    return app;
  }
```

Так я же не зря тебе написал добавить `return app`; Здесь ты делаешь app глобальным и в случае добавления флага `--parallel` для увечичения скорости, все сломается. Лучше этот app делать либо в каждом кейсе, либо в ближайшем `beforeEach`

```ts
desribe('root, () => {
  async function createApp() {
    return app;
  }
  describe('params, () => {
    let app: INestApplication;
    beforeEach(async () => {
      app = await createApp();
    })
  })
})
```